pkg/k8s/errors.go

```diff
@@ -233,3 +233,17 @@ func (e *ImmutableFieldsMutationError) Error() string {
 func NewImmutableFieldsMutationError(immutableFields []string) *ImmutableFieldsMutationError {
 	return &ImmutableFieldsMutationError{immutableFields}
 }
+
+type UpdateNotSupportedError struct {
+	resourceGVK    schema.GroupVersionKind
+	namespacedName types.NamespacedName
+}
+
+func (e *UpdateNotSupportedError) Error() string {
```

Comment:
We need code level guarantee that this is only called in `Update`. Otherwise, I suggest we rename it to something else.